core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/resourcePolicies/NaiveResourceAllocator.scala

```diff
@@ -0,0 +1,192 @@
+package edu.uci.ics.amber.engine.architecture.scheduling.resourcePolicies
+
+import edu.uci.ics.amber.engine.architecture.deploysemantics.PhysicalOp
+import edu.uci.ics.amber.engine.architecture.scheduling.Region
+import edu.uci.ics.amber.engine.architecture.scheduling.config.ChannelConfig.generateChannelConfigs
+import edu.uci.ics.amber.engine.architecture.scheduling.config.LinkConfig.toPartitioning
+import edu.uci.ics.amber.engine.architecture.scheduling.config.WorkerConfig.generateWorkerConfigs
+import edu.uci.ics.amber.engine.architecture.scheduling.config.{
+  LinkConfig,
+  OperatorConfig,
+  ResourceConfig
+}
+import edu.uci.ics.amber.engine.common.virtualidentity.PhysicalOpIdentity
+import edu.uci.ics.amber.engine.common.workflow.PhysicalLink
+import edu.uci.ics.texera.web.model.jooq.generated.tables.pojos.WorkflowRuntimeStatistics
+import edu.uci.ics.texera.workflow.common.workflow.{PartitionInfo, PhysicalPlan}
+
+import scala.collection.mutable
+
+/**
+  * Allocates resources for a given region and its operators.
+  *
+  * @param region The region for which to allocate resources.
+  * @return A tuple containing:
+  *         1) A new Region instance with new configuration.
+  *         2) An estimated cost of the workflow with the new configuration,
+  *         represented as a Double value (currently set to 0, but will be
+  *         updated in the future).
+  * Notes:
+  * 1. If the count of in running historical records for one operator are
+  *    less than 3, there is no need to allocate the workers, because it is
+  *    too fast. ( < 3, the first in running record might has warmup. So we
+  *    need at least one useful in running record to get the speed/throughput)
+  * 2. The historical data needs to have the same version with the current
+  *    workflow.
```

Comment:
the logic is you retrieve the historical data according to the current execution's workflow version. If there is no matching record, then there is no historical data.